src/aap_eda/api/views/activation.py

```diff
@@ -79,6 +83,14 @@ def create(self, request):
         )
         serializer.is_valid(raise_exception=True)
 
+        # If we're expected to run this activation we need redis
```

Comment:
Should we use the `is_enabled` field validator to check this? something like following in the serializer file?:
```
is_enabled = serializers.BooleanField(
    validators=[validators.check_if_redis_is_available],
)
```

Reply:
@hsong-rh I view a distinction between receiving a valid value and not being able to act on that value because Redis isn't available.

We could get the same functional result using a validator but, as described, that would require embedding system state checks into what is ostensibly a field value check.  If we were to opt to do it via the serializer I think a better choice would be to incorporate it as part of the validate() as that already takes a more holistic approach.

That having been said there are other usages (e.g., enable) where the decision needs to be made by the view and there's a strong development/maintenance argument to be made that the determination should be performed at a consistent level (in this case, views) in the processing.

Reply:
@jshimkus-rh Maybe `validate()` is a better choice. The logic in the view should be as simple as possible; all validation should happen in the serializer. That's my understanding.

Reply:
👌